### PR TITLE
Upgrade to tendermint 0.35.8 + builtin application

### LINKED ISSUE #5249: Upgrade Tendermint to 0.35

### LINKED ISSUE #5579: Migrate to builtin tendermint
Stop using a socket server and run tendermint and vega as part of the same binary.

### PR DESCRIPTION
**Pull request by jeremyletang**
close #5579  
close #5249

### CODE REVIEW

File: blockchain/nullchain/nullchain.go [hunk 1]
```diff
@@ -299,8 +298,8 @@ func (n *NullBlockchain) GetChainID(context.Context) (string, error) {
 
 func (n *NullBlockchain) GetStatus(context.Context) (*tmctypes.ResultStatus, error) {
 	return &tmctypes.ResultStatus{
-		NodeInfo: p2p.DefaultNodeInfo{
-			Version: "0.34.12",
+		NodeInfo: tmtypes.NodeInfo{
+			Version: "0.35.6",
```
Comment (ze97286) [inline blockchain/nullchain/nullchain.go:302]: 0.35.8

**Comment by wwestgarth** [comment]
I have a feeling the TM upgrade will cause the snapshot system-test to look like it failed because of the whole back-filling thing. So I've prepped a branch ready to go with a particular check switched off so whenever this PR is ready the merge won't get blocked by the test failing: https://github.com/vegaprotocol/system-tests/compare/change-snapshot-test?expand=1

**Comment by wwestgarth** [comment]
> > I have a feeling the TM upgrade will cause the snapshot system-test to look like it failed because of the whole back-filling thing. So I've prepped a branch ready to go with a particular check switched off so whenever this PR is ready the merge won't get blocked by the test failing: https://github.com/vegaprotocol/system-tests/compare/change-snapshot-test?expand=1
> 
> I have a feeling your feeling is correct, but can we not just set the evidence size to like 10-100 and test?
> 
> specifically these:
> 
> ```
> "evidence": {
>         "max_age_num_blocks": "100000",
>         "max_age_duration": "172800000000000",
>         "max_bytes": "1048576"
>       },
> ```

Yup you could do that aswell.

**Comment by jeremyletang** [comment]
```json
{
  "system-tests": "feature/tendermint-builtin-35.8"
}
```

**Review by ze97286 — APPROVED** [review]

**Pull request merged**